packages/extension-api/src/extension-api.d.ts

```diff
@@ -2692,4 +2692,60 @@ declare module '@podman-desktop/api' {
       metadata?: ImageCheckerProviderMetadata,
     ): Disposable;
   }
+
+  export interface ContainerNavigation {
+    // Navigate to the Containers page in Podman-Desktop
+    viewContainers(): Promise<void>;
+    // Navigate to the Container page in Podman-Desktop
+    viewContainer(id: string): Promise<void>;
+    // Navigate to the Container logs tab in Podman-Desktop
+    viewContainerLogs(id: string): Promise<void>;
+    // Navigate to the Container inspect tab in Podman-Desktop
+    viewContainerInspect(id: string): Promise<void>;
+    // Navigate to the Container terminal tab in Podman-Desktop
+    viewContainerTerminal(id: string): Promise<void>;
```

Comment:
I think id might not be enough as you can have the same container's name between docker engine and podman engine (it requires engineId) as well

As we handle ContainerInfo or ImageInfo object etc we could accept this object directly ?
```
navigateToContainerDetails(containerInfo: ContainerInfo)
navigateToContainerDetails(engineId: string, id: string)
```

Reply:
> I think id might not be enough as you can have the same container's name between docker engine and podman engine (it requires engineId) as well
> 
> As we handle ContainerInfo or ImageInfo object etc we could accept this object directly ?
> 
> ```
> navigateToContainerDetails(containerInfo: ContainerInfo)
> navigateToContainerDetails(engineId: string, id: string)
> ```

How does this would translate in terms of path ? I used the `App.svelte` as a source of requirements for the routes

https://github.com/containers/podman-desktop/blob/caf13ef86dbed5a3f775963827ea2bdfc4eb54ee/packages/renderer/src/App.svelte#L95-L97

Reply:
it depends on what we're calling 'id' from consumer POV

for example on CLI you can do : `podman stop container-name` or `podman stop container-id`

so it means client won't be able to call `navigateToContainer('container-name')` but always use the `('container-shaID')`


if we accept ContainerInfo object it means we don't care much on what we authorize from id

also, what happens on the API if the id does not exists, should it return a boolean flag ?

Reply:
> it depends on what we're calling 'id' from consumer POV

For examples the method `createContainer` returns `ContainerCreateResult` which only contains an `id: string` property. On the other side, the ContainerInfo is only returned by the `listContainers` methods. If we only support from ContainerInfo an extension would have first to createContainer, then listContainers, find on id, navigateToContainer then.

Do we want to support both ? `navigateToContainer(shaId: string)` and `navigateToContainer(info: ContainerInfo)` ?

> Supporting both _seems_ to need different method name.

Reply:
> also, what happens on the API if the id does not exists, should it return a boolean flag ?

We might either throw an error ?

Reply:
for sure we can use same method name

but for now we can use only 'id' being the sha id (you can't give a container's name)

Reply:
> > also, what happens on the API if the id does not exists, should it return a boolean flag ?
> 
> We might either throw an error ?

Or do we even want to check that ? This might be the extension responsibility ? And we simply display not found on the renderer if the id does not exist